ekncontent/ekncontent/eknc-utils.c

```diff
@@ -370,6 +374,93 @@ eknc_get_data_dir (const gchar *app_id)
   return NULL;
 }
 
+static GList *
+databases_dirs_from_metadata (const gchar *flatpak_path, const gchar *app_id)
+{
+  GList *list = NULL;
+  g_autoptr(GKeyFile) metakey = NULL;
+  g_auto(GStrv) groups = NULL;
+  g_autofree gchar *metadata_path = NULL;
+  g_autofree gchar *runtime = NULL;
+  g_auto(GStrv) components = NULL;
+  gchar *arch = NULL;
+
+  metadata_path = g_build_filename (flatpak_path, "flatpak",
+                                    "app", app_id,
+                                    "current", "active", "metadata",
+                                    NULL);
+  metakey = g_key_file_new ();
+  if (!g_key_file_load_from_file (metakey, metadata_path, G_KEY_FILE_NONE, NULL))
+    return NULL;
+
+  runtime = g_key_file_get_string (metakey, "Application", "runtime", NULL);
+  components = g_strsplit (runtime, "/", 3);
+  arch = components[1];
+
+  groups = g_key_file_get_groups (metakey, NULL);
+  for (int i = 0; groups[i] != NULL; i++)
+    {
+      g_autoptr(GFile) extension_dir = NULL;
+      gchar *extension_name = NULL;
+      g_autofree gchar *extension_version = NULL;
+      g_autofree gchar *extension_path = NULL;
+
+      if (!g_str_has_prefix (groups[i], "Extension"))
+        continue;
+
+      extension_name = g_strstr_len (groups[i], -1, app_id);
+      if (extension_name == NULL)
+        continue;
+
+      extension_version = g_key_file_get_string (metakey, groups[i], "version", NULL);
+      if (extension_version == NULL)
+        continue;
+
+      extension_name = g_strstrip (extension_name);
+      extension_path = g_build_filename (flatpak_path, "flatpak",
+                                         "runtime", extension_name,
+                                         arch, extension_version,
+                                         "active", "files",
+                                          NULL);
+      extension_dir = g_file_new_for_path (extension_path);
+      if (!g_file_query_exists (extension_dir, NULL))
+        continue;
+
+      list = g_list_prepend (list, g_file_dup(extension_dir));
```

Comment:
nit: you could increment the reference count with g_object_ref() instead of creating a new object